src/main/java/org/breedinginsight/brapps/importer/daos/BrAPITrialDAO.java

```diff
@@ -123,8 +135,10 @@ private List<BrAPITrial> processExperimentsForDisplay(List<BrAPITrial> trials, S
         return displayExperiments;
     }
 
-    public Optional<BrAPITrial> getTrialByDbId(String trialDbId, Program program) throws ApiException {
-        List<BrAPITrial> trials = getTrialsByDbIds(List.of(trialDbId), program);
+    public Optional<BrAPITrial> getTrialByDbId(UUID programId, UUID trialDbId) throws ApiException, DoesNotExistException {
```

Comment:
I think this is misleading now.  Originally the method was fetching trials by the actual `trialDbId` created by the BrAPI server holding the data (i.e. BreedBase).  I think there should be a new method of `getTrialById` that looks like this, and revert this method to what it originally was.

Reply:
done